### PR TITLE
Adds driver for MP2660

### PR DESCRIPTION
**Pull request by mmarkwort**

### CODE REVIEW

File: components/mp2660/.eil.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: mp2660
+description: Driver for MP2660 5V USB, 500mA, I2C-Controlled Linear Charger with Power Path Management for Single-Cell Li-Ion Battery
+version: 1.0.0
+groups:
+  - misc
```
Comment (UncleRus) [inline components/mp2660/.eil.yml:5]: the group should be `battery`, not `misc`

File: components/mp2660/mp2660.h [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+#define MP2660_POWER_ON_CONFIG_REG_DEFAULT 0x4
+#define MP2660_CURENT_CONTROL_REG_DEFAULT 0xE
+#define MP2660_PRECHARGE_TERMINATION_CURRENT_REG_DEFAULT 0x4A
+#define MP2660_CHARGE_VOLTAGE_CONTROL_REG_DEFAULT 0xA3
+#define MP2660_CHARGE_TERMINATION_TIMER_CONTROL_REG_DEFAULT 0x4A
+#define MP2660_MISC_OPERATION_CONTROL_REG_DEFAULT 0xB
+#define MP2660_SYSTEM_STATUS_REG_DEFAULT 0x0
+#define MP2660_FAULT_REG_DEFAULT 0x0
+
+// Input Source Control Register/Address: 00h (Default: 0100 1111)
```
Comment (UncleRus) [inline components/mp2660/mp2660.h:63]: All enums, structures, unions and their fields must be commented in Doxygen style.
```C
/**
 * Super-duper enum
 */
typedef enum
{
     ENUM_VAL_0,  //!< Important enum value
     ENUM_VAL_1,  //!< Not so important enum value
} my_enum_t;

/**
 * Config structure
 */
typedef struct
{
    int foo;   //!< Comment to field
} my_struct_t;
```

and so on

File: components/mp2660/.eil.yml (hunk 1 above)
Comment (mmarkwort) [reply components/mp2660/.eil.yml]: Done. I searched for charger, not for battery :-)

File: components/mp2660/mp2660.h (hunk 2 above)
Comment (mmarkwort) [reply components/mp2660/mp2660.h]: Hoping, I commented now all correctly

File: components/mp2660/mp2660.c [hunk 3]
```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2017 Gunar Schorcht <https://github.com/gschorcht>
```
Comment (UncleRus) [inline components/mp2660/mp2660.c:2]: Invalid copyrights

File: components/mp2660/mp2660.h [hunk 4]
```diff
@@ -0,0 +1,424 @@
+/*
+ * Copyright (c) 2017 Gunar Schorcht <https://github.com/gschorcht>
```
Comment (UncleRus) [inline components/mp2660/mp2660.h:2]: Invalid copyrights

File: components/mp2660/mp2660.c (hunk 3 above)
Comment (mmarkwort) [reply components/mp2660/mp2660.c]: Copy & Paste fixed

File: components/mp2660/mp2660.h (hunk 4 above)
Comment (mmarkwort) [reply components/mp2660/mp2660.h]: Copy & Paste fixed

File: examples/mp2660/default/main/main.c [hunk 5]
```diff
@@ -0,0 +1,227 @@
+#include <inttypes.h>
```
Comment (UncleRus) [inline examples/mp2660/default/main/main.c:1]: Please format source with `clang-format`. Repository contains the clang-format settings

**Comment by UncleRus** [comment]
You also need to make appropriate changes to the documentation (see `docs/source`)

File: examples/mp2660/default/main/main.c (hunk 5 above)
Comment (mmarkwort) [reply examples/mp2660/default/main/main.c:1]: Done.

**Comment by mmarkwort** [comment]
Added MP2660 to *.rst files

**Comment by UncleRus** [comment]
Thank you!

**Pull request merged**